webapp/app/forms/steps/steuerlotse_step.py

```diff
@@ -115,34 +119,47 @@ class InputForm(SteuerlotseBaseForm):
         pass
 
     def __init__(self, endpoint, header_title, stored_data=None, overview_step=None, default_data=None, prev_step=None,
-                 next_step=None, session_data_identifier='form_data'):
+                 next_step=None, session_data_identifier='form_data', update_data=False, form_data=None,
+                 data_is_valid=False):
         super().__init__(endpoint, header_title, stored_data, overview_step, default_data, prev_step, next_step,
-                         session_data_identifier=session_data_identifier)
+                         session_data_identifier=session_data_identifier, update_data=update_data, form_data=form_data,
+                         data_is_valid=data_is_valid)
         # TODO rename this to form_class once MultiStepFlow is obsolete
         self.form = self.InputForm
 
     @classmethod
-    def create_form(cls, request, prefilled_data):
+    def create_form(cls, form_data, prefilled_data):
         # If `form_data` is present it will always override `data` during
         # value binding. For `BooleanFields` an empty/missing value in the `form_data`
         # will lead to an unchecked box.
-        form_data = request.form
         if len(form_data) == 0:
             form_data = None
 
         return cls.InputForm(form_data, **prefilled_data)
-    
+
     @classmethod
-    def update_data(cls, stored_data):
-        if request.method == 'POST':
-            form = cls.create_form(request, prefilled_data=stored_data)
-            if form.validate():
-                stored_data.update(form.data)
+    def update_data(cls, stored_data, data_to_update):
+        """
+            This updates the stored_data with the updated_data.
+
+            :param stored_data: The stored session data
+            :param data_to_update: The data with which the stored data
+            should be updated. Note: the stored data should already be validated.
+        """
+        stored_data.update(data_to_update)
```

Comment:
I'm confused, sorry.
Maybe rename `data_to_update` -> `data_to_update_with`
`updated_data` -> `data_to_update_with`
`Note: the stored data should already be validated.` -> `Note: the data_to_update_with should already be validated.`